api/v1alpha3/azurecluster_conversion.go

```diff
@@ -150,6 +154,12 @@ func Convert_v1alpha3_AzureClusterSpec_To_v1beta1_AzureClusterSpec(in *AzureClus
 		return err
 	}
 
+	// copy AzureClusterCommon fields
+	out.SubscriptionID = in.SubscriptionID
```

Comment:
seems strange that we have to do this, do you know why it's necessary?

Reply:
This is due to conversion gen not being able to recognize embedded struct fields, so had to write manual conversion for all fields that are moved to common structs. Admittedly, this doesn't look good, but we won't have to do this for future api versions. https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/a6d75b76d20a87f552e709b668afe4649cca21ad/api/v1alpha3/zz_generated.conversion.go#L671